projects/gloo/pkg/translator/listener_subsystem.go

```diff
@@ -0,0 +1,91 @@
+package translator
+
+import (
+	validationapi "github.com/solo-io/gloo/projects/gloo/pkg/api/grpc/validation"
+	v1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	"github.com/solo-io/gloo/projects/gloo/pkg/plugins"
+	"github.com/solo-io/gloo/projects/gloo/pkg/utils"
+)
+
+// The Listener subsystem (https://www.envoyproxy.io/docs/envoy/latest/intro/life_of_a_request.html?highlight=listener#high-level-architecture)
+// handles downstream request processing.
+// Gloo sends resources to Envoy via xDS. The components of the Listener subsystem that Gloo configures are:
+// 1. Listeners
+// 2. RouteConfiguration
+// Given that Gloo exposes a variety of ListenerTypes (HttpListener, TcpListener, HybridListener), and each of these types
+// affect how resources are generated, we abstract those implementation details behind abstract translators.
+// The ListenerSubsystemTranslatorFactory returns a ListenerTranslator and RouteConfigurationTranslator for a given Gloo Listener
+type ListenerSubsystemTranslatorFactory struct {
+	plugins             []plugins.Plugin
+	proxy               *v1.Proxy
+	sslConfigTranslator utils.SslConfigTranslator
+}
+
+func NewListenerSubsystemTranslatorFactory(
+	plugins []plugins.Plugin,
+	proxy *v1.Proxy,
+	sslConfigTranslator utils.SslConfigTranslator,
+) *ListenerSubsystemTranslatorFactory {
+	return &ListenerSubsystemTranslatorFactory{
+		plugins:             plugins,
+		proxy:               proxy,
+		sslConfigTranslator: sslConfigTranslator,
+	}
+}
+
+func (l *ListenerSubsystemTranslatorFactory) GetListenerSubsystemTranslators(listener *v1.Listener, listenerReport *validationapi.ListenerReport) (
```

Comment:
Question: How would we feel about GetTranslators as this is already an instance of ListenerSubststemTranslatorFactory?
```suggestion
func (l *ListenerSubsystemTranslatorFactory) GetTranslators(listener *v1.Listener, listenerReport *validationapi.ListenerReport) (
```